app/test_utils.go

```diff
@@ -0,0 +1,42 @@
+package app
+
+import (
+	"encoding/json"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/log"
+	dbm "github.com/tendermint/tm-db"
+)
+
+func setup(withGenesis bool, invCheckPeriod uint) (*App, GenesisState) {
+	db := dbm.NewMemDB()
+	encCdc := MakeEncodingConfig()
+	newApp := New(log.NewNopLogger(), db, nil, true, map[int64]bool{}, DefaultNodeHome, invCheckPeriod, encCdc, simapp.EmptyAppOptions{})
+	if withGenesis {
+		return newApp, NewDefaultGenesisState(encCdc.Marshaler)
+	}
+	return newApp, GenesisState{}
+}
+
+// Setup initializes a new SimApp. A Nop logger is set in SimApp.
+func Setup(isCheckTx bool) *App {
```

Comment:
@alessio any opinions on using a app setup, similar to how simapp is used in sdk?